app/src/main/java/fr/free/nrw/commons/upload/UploadController.java

```diff
@@ -129,8 +130,18 @@ public void startUpload(String title, Uri mediaUri, String description, String m
     @SuppressLint("StaticFieldLeak")
     public void startUpload(final Contribution contribution, final ContributionUploadProgress onComplete) {
         //Set creator, desc, and license
-        if (TextUtils.isEmpty(contribution.getCreator())) {
-            contribution.setCreator(sessionManager.getCurrentAccount().name);
+
```

Comment:
We add SharedPreferences to CommonsApplicationModule and use @Inject to use them. You can see examples in code. This doesn't change something functional but it would be better if you can use in that way, to preserve code style.